packages/render-dedupe-check-issues/src/index.ts

```diff
@@ -0,0 +1,56 @@
+import { type ResolutionChange, type DedupeCheckIssues, type ResolutionChangesByAlias, type SnapshotsChanges } from '@pnpm/types'
+import archy from 'archy'
+import chalk from 'chalk'
+
+export function renderDedupeCheckIssues (dedupeCheckIssues: DedupeCheckIssues) {
+  const importersReport = report(dedupeCheckIssues.importerIssuesByImporterId)
+  const packagesReport = report(dedupeCheckIssues.packageIssuesByDepPath)
+
+  const lines = []
+  if (importersReport != null) {
+    lines.push(chalk.blueBright.underline('Importers'))
+    lines.push(importersReport)
+    lines.push('')
+  }
+  if (packagesReport != null) {
+    lines.push(chalk.blueBright.underline('Packages'))
+    lines.push(packagesReport)
+    lines.push('')
+  }
+
+  return lines.join('\n')
+}
+
+function report (snapshotChanges: SnapshotsChanges): string | null {
+  if (countChangedSnapshots(snapshotChanges) === 0) {
+    return null
+  }
+
+  return [
+    ...Object.entries(snapshotChanges.updated).map(([alias, updates]) => archy(toArchy(alias, updates))),
+    ...snapshotChanges.added.map((id) => `${chalk.green('+')} ${id}`),
+    ...snapshotChanges.removed.map((id) => `${chalk.red('-')} ${id}`),
+  ].join('\n')
+}
+
+function toArchy (name: string, issue: ResolutionChangesByAlias): archy.Data {
+  return {
+    label: name,
+    nodes: Object.entries(issue).map(([alias, change]) => toArchyResolution(alias, change)),
+  }
+}
+
+function toArchyResolution (alias: string, change: ResolutionChange): archy.Data {
+  switch (change.type) {
+  case 'added':
+    return { label: `${chalk.green('+')} ${alias} ${chalk.gray(change.next)}` }
+  case 'removed':
+    return { label: `${chalk.red('-')} ${alias} ${chalk.gray(change.prev)}` }
+  case 'updated':
+    return { label: `${alias} ${chalk.red(change.prev)} ${chalk.gray('→')} ${chalk.green(change.next)}` }
```

Comment:
Some terminals may not support unicode. If that's a concern, we can switch `→` to `->`.

Reply:
I am not sure. I think we currently don't print any unicode but I wanted to do this: https://github.com/pnpm/pnpm/pull/6226/files

Reply:
I saw https://github.com/sindresorhus/is-unicode-supported. Based on the description there, it sounds like there's very few terminals in practice that don't support Unicode. It's probably just the Windows Command Prompt, although I would bet most users are using Windows Terminal.

Reply:
what about CI?

Reply:
The `is-uncode-supported` package seems to assume most CI solutions support Unicode.

https://github.com/sindresorhus/is-unicode-supported/blob/c80c691dde9e2fcfe3996810858c6672c8f35ad9/index.js#L8

Reply:
There might be a few users on the Window command prompt. We can add `is-unicode-supported` and fallback to `...` or `->`?

Reply:
I don't think that is a good idea as that changes the length

Reply:
Oh I see. The PR you linked to touches code that's fairly sensitive to length. Good point.